src/frontend/src/optimizer/rule/agg_project_merge_rule.rs

```diff
@@ -24,31 +24,33 @@ pub struct AggProjectMergeRule {}
 impl Rule for AggProjectMergeRule {
     fn apply(&self, plan: PlanRef) -> Option<PlanRef> {
         let agg = plan.as_logical_agg()?;
-        let (mut agg_calls, agg_group_keys, grouping_sets, input) = agg.clone().decompose();
```

Comment:
I am wondering if it is a good pattern... If the developer does not care about some value, he can be not aware of them. All fields in the `Agg<PlanRef>` are public fields and can be modified just like an `AggBuilder`...

Reply:
In my mind, `decompose` should expose as many fields as possible to avoid developers forgetting some hidden fields. Of course, they can decide to ignore or not by `let (interested_field1, interested_field2, _) = decompose()`